File: backend/member-service/src/main/java/com/example/memberservice/global/lock/LockParam.java
package com.example.memberservice.global.lock;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 이 파라미터의 값이 락 키가 된다. {@link Lockable} 이면 getKey() 를 쓴다.
 */
@Target(ElementType.PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
public @interface LockParam {
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/global/lock/LockParam.kt
package com.example.memberservice.global.lock

/**
 * 이 파라미터의 값이 락 키가 된다. [Lockable] 이면 key 를 쓴다.
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class LockParam
